front_end/src/backend/downloads.js

```diff
@@ -0,0 +1,149 @@
+/**
+ * Copyright 2022 Google LLC
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// In these functions the 'fileName' parameter letiables refer to filenames of '2mib.txt' '64mib.txt' and '256mib.txt'
+const REGIONS_JSON_STRING = `{
+    "northamerica-northeast1":"Montréal",
+    "northamerica-northeast2":"Toronto",
+    "us-central1":"Iowa",
+    "us-east1":"South Carolina",
+    "us-east4":"Northern Virginia",
+    "us-east5":"Columbus",
+    "us-south1":"Dallas",
+    "us-west1":"Oregon",
+    "us-west2":"Los Angeles",
+    "us-west3":"Salt Lake City",
+    "us-west4":"Las Vegas",
+    "southamerica-east1":"São Paulo",
+    "southamerica-west1":"Santiago",
+    "europe-central2":"Warsaw",
+    "europe-north1":"Finland",
+    "europe-southwest1":"Madrid",
+    "europe-west1":"Belgium",
+    "europe-west2":"London",
+    "europe-west3":"Frankfurt",
+    "europe-west4":"Netherlands",
+    "europe-west6":"Zürich",
+    "europe-west8":"Milan",
+    "europe-west9":"Paris",
+    "asia-east1":"Taiwan",
+    "asia-east2":"Hong Kong",
+    "asia-northeast1":"Tokyo",
+    "asia-northeast2":"Osaka",
+    "asia-northeast3":"Seoul",
+    "asia-south1":"Mumbai",
+    "asia-south2":"Delhi",
+    "asia-southeast1":"Singapore",
+    "asia-southeast2":"Jakarta",
+    "australia-southeast1":"Sydney",
+    "australia-southeast2":"Melbourne"
+}`;
+const REGIONS_MAP = JSON.parse(REGIONS_JSON_STRING);
```

Comment:
Is there a reason to treat the regions as a JSON string as opposed to just creating an object and avoiding the parse operation here?

Reply:
I would just create an object here and avoid the extraneous call to convert (JSON.parse) from a string which appears unneeded in this context.